packages/block-library/src/comments-pagination-next/index.php

```diff
@@ -37,7 +37,7 @@ function render_block_core_comments_pagination_next( $attributes, $content, $blo
 		$label .= $pagination_arrow;
 	}
 
-	$next_comments_link = get_next_comments_link( $label, $max_page );
+	$next_comments_link = get_next_comments_link( $label, $max_page, $comment_vars['paged'] );
```

Comment:
As mentioned in https://github.com/WordPress/wordpress-develop/pull/7275#issuecomment-2327066295, we need to add some safeguards to make sure that we're not passing this extra argument if `get_next_comments_link()` -- as provided by whatever WordPress version is running -- doesn't support the additional parameter yet.

To do so, I think our best course of action is PHP's [`ReflectionFunction`](https://www.php.net/manual/en/class.reflectionfunction.php):

Something like (untested):

```suggestion
	$get_next_comments_link_reflection = new ReflectionFunction( 'get_next_comments_link' );

	if ( $get_next_comments_link_reflection->getNumberOfParameters() >= 3 ) {
		$next_comments_link = get_next_comments_link( $label, $max_page, $comment_vars['paged'] );
	} else {
		$next_comments_link = get_next_comments_link( $label, $max_page );
	}
```

and accordingly for `get_previous_comments_link`.

Reply:
Do we really need to add this safety check? If I am not mistaken, when passing the third argument to a function that only accepts two, PHP will just ignore it, right? At least, that's what I can see from my testing.

Reply:
There are functions like [func_get_args](https://www.php.net/manual/en/function.func-get-args.php) that allow to handle any number of params, so as far as I testeddditional param that it, adding an a isn't handled works just fine.

Reply:
Ah, fair enough! I was assuming that PHP would at least issue a warning or a notice if strict error reporting was enabled, but it seems that that isn't the case. (Here's a related [SO thread](https://stackoverflow.com/questions/7928707/why-does-php-not-throw-an-error-when-i-pass-too-many-parameters-to-a-function).)

Reply:
Yes, it fatals on built-in functions I tested, but not on custom functions 🤷🏻

Reply:
I tested it directly with `get_next_comments_link` in a branch without the changes and it doesn't seem to break.